angular-client/src/components/battery/battery.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, Input } from '@angular/core';
+import Theme from 'src/services/theme.service';
+
+@Component({
+  selector: 'battery',
+  templateUrl: 'battery.component.html',
+  styleUrls: ['battery.component.css']
+})
+export class BatteryPercentageComponent {
+  @Input() props!: { percentage: number; height: number; width: number };
```

Comment:
eh we dont really do it like this anywhere else, we create seperate inputs for all of these which i think you should do here as well

Reply:
sounds good